robottelo/ui/hosts.py

```diff
@@ -57,6 +57,22 @@ def _configure_hosts_parameters(self, parameters_list):
                 self.click(
                     common_locators['entity_select_list'] % parameter_value)
                 continue
+            elif parameter_name == 'Puppet Environment':
+                # Make sure 'inherit' button is not pressed before selecting
+                # puppet environment, as otherwise 'Puppet Environment'
+                # dropdown will be disabled
+                # Unfortunately, 'inherit' button doesn't have any attribute
+                # providing its state, checking whether 'Puppet Environment'
+                # dropdown is enabled instead
+                self.click(param_locator)
+                dropdown = self.wait_until_element(
```

Comment:
Right, that `select_list_search_box` is only visible when you open the drop-down. And it can be only opened when it's enabled.